### DISCUSSION TITLE
Block Highlighting

### DISCUSSION THREAD

**Discussion by izhangzhihao** [izhangzhihao/intellij-rainbow-brackets#2644] [Show and tell]
**The Highlighting effect will follow your caret position, the effect will happen without any shortcuts**

* Setting page
<img width="1142" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/4705e0f7-7b48-4d20-8dcb-ada367cbb210">

* Java Current Scope Highlight
![Java Current Scope Highlight](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/6e56c6ee-61e1-450d-aebe-64d48172ffcc)

* Java Current Restrain Scope Highlight
![Java Current Restrain Scope Highlight](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/c3221762-b6e8-427d-96ed-6a9d3e59774a)

* Python Current Restrain Scope Highlight
![Python Current Restrain Scope Highlight](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/d65f72ab-bdc3-40e9-ba27-84644aa77f61)

* Python Current Scope Highlight With Rainbow Python Keywords enabled
![Python Current Scope Highlight With Rainbow Keywords](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/3dd7c703-5187-478f-b78a-ed5c99f0871f)

**Comment by kylegrieder** [comment]
I love this! I've been wanting this feature ever since VSCode started including it by default. I'm hoping for a highlight which more closely resembles what is in VSCode: something a little less flashy than "Highlight" but a little more than "Restrain". I've attached a picture of an example. Really I'd just love the additional border surrounding the scope, not the background highlight. Could this be added as a third option?

![CleanShot 2023-07-14 at 10 51 11@2x](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/25424229/de43343d-2b39-4bc1-9648-9b870097c509)
![CleanShot 2023-07-14 at 10 51 29@2x](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/25424229/30cbe6a3-8be6-4e1d-847b-3bcc86a9eae8)

**Comment by izhangzhihao** [comment]
<img width="962" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/b93e8833-e3e2-4cb1-abe4-bb23f2cf6dee">




https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/ad5a221c-0126-4b2d-a4c2-6dd355b34a49

![Broder highlight](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/8f0ee97c-6f0a-4766-8493-913300517a5d)

**Comment by kylegrieder** [comment]
Just installed it and it is working great so far!

**Comment by rioscode** [comment]
I should read the "What's new in Rain.." more often, love this feature. :-).

**Comment by demensky** [comment]
Feature is very cool!

Is there a way to make the border colors more transparent? The current ones are too distracting.

**Reply by izhangzhihao** [reply]
The latest release supported that, please have a try~
And BTW it will continue to enhance in the next release.

**Comment by kylegrieder** [comment]
Absolutely love this feature. 

I would love to see it continue to enhance, specifically to include an option to make the borders more like vscode, meaning the border would only exist on the indentation line, and along the top part of the scope. another way to explain this would be to remove the bottom and right borders for each scope.

thanks for this update!

**Reply by izhangzhihao** [reply]
Great idea, I'm still trying to find how to achieve this.

**Reply by pochat** [reply]
Hi all,

I installed the 30-day trial but can't see the borders. What am I missing?

Thank you.

![rb](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/48400262/feffffa3-4acb-4d9b-9383-9b4b030aaca1)
![rb4](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/48400262/a4411c05-56c9-4e0a-8266-188972d4f0eb)
![rb3](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/48400262/45c2c76c-b5a1-4c03-a56f-151146aa9918)
![rb2](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/48400262/6878b097-72de-4ad0-a2df-9d309bda7ef9)

**Reply by izhangzhihao** [reply]
Hi, quick update here, Current border highlighting does not support Vue yet, I will take a try to support Vue in the future.

**Reply by pochat** [reply]
Hi @izhangzhihao,

I hope you are well and thank you for responding.

It will be great if your plugin supports Vue. I'll buy a license once it does.

Thank you.

**Reply by ashokgelal** [reply]
+1 for VueJS support

**Reply by izhangzhihao** [reply]
This feature will be available in the next release: 
<img width="934" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/9aaa545e-a47e-4ec8-82b6-5525cc1ff5bf">

**Reply by pochat** [reply]
Hi @izhangzhihao,

I hope you are well, and thank you for the update.

The screenshot looks great. Will you add the border highlight as well?

I'm looking forward to this update.

thank you.

**Reply by izhangzhihao** [reply]
> Hi @izhangzhihao,
> 
> 
> 
> I hope you are well, and thank you for the update.
> 
> 
> 
> The screenshot looks great. Will you add the border highlight as well?
> 
> 
> 
> I'm looking forward to this update.
> 
> 
> 
> thank you.

Yes, border highlighting will be available as well.

**Reply by ashokgelal** [reply]
I tested Rainbow Brackets for a couple of weeks mostly on VueJS projects and overall I like it and it works well except that for Single File Components it highlights everything within `<script></script>`, when your cursor is anywhere within it, which is a bit annoying. I would much prefer it only highlighted the current scope. I'm not sure if it is easy to fix or not but once that is taken care of I think I am ready to purchase a lifetime license. Happy to give you more details about the above issue if you want.

**Comment by izhangzhihao** [comment]
An enhancement already made for better support of Vue and JSX(it's just the case you mentioned). But I'm still waiting for next release of the 2024.1 series. Maybe 1 or 2 weeks later I Can make it publicly available .
________________________________

**Reply by ashokgelal** [reply]
That's great! Is it possible to side load right now somehow? I can test it and give some feedback.

**Reply by izhangzhihao** [reply]
pls email me izhangzhihao@hotmail.com

**Comment by luckypipitao** [comment]
请问一下clion中的c++是不是使用不了这个功能啊（我使用的是试用的30天）

**Reply by izhangzhihao** [reply]
Clion Nova is not supported yet. Clion Classic is supported.

**Reply by luckypipitao** [reply]
换成classic可以使用这个功能了，请问一下是否在之后的版本会适配Clion Nova

**Reply by izhangzhihao** [reply]
Clion Nova is still missing some fundamental modules, so IDE must implement those modules first.
